pkg/controller/direct/logging/logmetric_controller.go

```diff
@@ -150,6 +150,24 @@ func (a *logMetricAdapter) Create(ctx context.Context, u *unstructured.Unstructu
 	if filter == "" {
 		return fmt.Errorf("filter is empty")
 	}
+	if a.desired.Spec.LoggingLogBucketRef != nil {
+		bucketName := a.desired.Spec.LoggingLogBucketRef.External
+		if bucketName != "" {
+			// todo acpana support resolving non external ref for loggingLogBucketRef
+
+			// validate the the bucket ref external is well formatted
+			// eg: projects/my-project/locations/global/buckets/my-bucket
+			parts := strings.Split(bucketName, "/")
```

Comment:
I think we will want some general util functions to validate and parse the full resource name (basically the "parent + resource" form), like
- `projects/{PROJECT}/locations/{LOCATION_ID}/{RESOURCE_NAMES}/{RESOURCE_ID}`
- `projects/{PROJECT}/{RESOURCE_NAMES}/{RESOURCE_ID}`
and probably some others. 

How about a new file `pkg/controller/direct/parse/resourceid.go` for that purpose?  Open to other dir/file naming ideas.

Reply:
There has been some chatter about this in the past. I think we can move towards having some `parser.go`.  I can follow on this PR with more usage (outside of this controller). cc @justinsb